### DISCUSSION TITLE
Backup system performs a backup every 2 minutes rendering Inventree unusable

### DISCUSSION THREAD

**Discussion by MR-DOS** [inventree/InvenTree#4179] [Q&A]
Hi, I have been using Inventree for quite a while in a docker-compose configuration.

Now, I have upgraded from 0.8.3 to 0.9.2. However, I cannot start the server at all. It keeps on spawning workers and killing them and creating SQL backups every 2 minutes (exactly, so probably not a coincidence). The SQL backup keeps all my 4 cores on a server loaded to 100% and it's not able to do anything else. Is this expected behaviour of the new backup system or did I do anything wrong?
![image](https://user-images.githubusercontent.com/29711626/211203656-77ae837a-0ff7-4b43-bd74-daa06c49a719.png)

**Comment by MR-DOS** [comment]
Uh, I'm probably running into the same or similar issue as this:
https://github.com/inventree/InvenTree/issues/4112

**Comment by MR-DOS** [comment]
OK, so I can run it by restricting Gunicorn workers and background workers both to 1 (now on version 0.9.1, will try going back up to 0.9.2). Then it finally manages to start up and Inventree is usable again.
But it still keeps on generating backups every 2 minutes during startup of the server for some reason which consumes a lot of resources.

Could it be that the reason why I cannot start Inventree with more workers is that it is trying to make backup during startup of the server, preventing proper startup, running out of time and then it all runs in a circle?

Now that it started up, it generated backup 3 times in a row until it finally decided it has enough backups...

**Comment by MR-DOS** [comment]
OK, so after shutdown, going up to 0.9.2 and new startup, it's not doing the backups again this time.

An idea that comes to mind: The backup task should be postponed until some uptime of Inventree server is reached so as not to block proper startup of the server by the backup.

For anyone else running into this issue on slow processors like the Atom N2800:

- Add this to your .env file:

```
INVENTREE_GUNICORN_WORKERS=1
INVENTREE_BACKGROUND_WORKERS=1
```

- Add this to your config.yaml:

```
background:
  workers: 1
  timeout: 90
```

Not sure what the timeout means since workers still seem to get killed after 30 seconds according to log.

**Reply by SchrodingersGat** [reply]
The timeout here refers to the background workers. The foreground process (gunicorn) seems to be the one timing out here, no?

**Comment by MR-DOS** [comment]
Also one problem from older versions happened to me again - the left navigation bar in Part view went missing and System Information was unavailable until I logged out, cleared cookies and site data and logged in again.

**Reply by matmair** [reply]
@MR-DOS this is intended behaviour, as discussed before: https://docs.inventree.org/en/latest/faq/#feature-x-is-not-visible-after-update

**Comment by matmair** [comment]
Please file problems as bugs - discussions are very hard to follow with several responses.

**Reply by MR-DOS** [reply]
At the time of writing, I wasn't sure it was a bug and not just problem on my end. However as can be seen, it already was an issue, closed but not resolved at all.
